src/config_reader.cpp

```diff
@@ -125,63 +108,43 @@ static char *StripWhitespace(char *first, char *last = nullptr)
 
 	assert(*last == '\0');
 
-	while (first < last && IsSpace(*first)) first++;
-	while (last > first + 1 && IsSpace(last[-1])) last--;
+	while (first < last && isspace(*first)) first++;
+	while (last > first + 1 && isspace(last[-1])) last--;
 	*last = '\0';
 	return first;
 }
 
-/**
- * Try to find the configuration file from the list of directories, and load it if found.
- * @param dir_list \c nullptr terminated list of directories to try.
- * @param fname Filename to look for.
- * @return Loading succeeded.
- */
-bool ConfigFile::LoadFromDirectoryList(const char **dir_list, const char *fname)
-{
-	while (*dir_list != nullptr) {
-		std::string fpath = std::string(*dir_list) + '/' + fname;
-		if (PathIsFile(fpath.c_str()) && this->Load(fpath.c_str())) {
-			return true;
-		}
-		dir_list++;
-	}
-	return false;
-}
-
 /**
  * Load a config file.
  * @param fname Filename to load.
- * @return Loading succeeded.
  * @todo [easy] Eliminate duplicate config sections.
  * @todo [easy] Strip whitespace around section names.
  */
-bool ConfigFile::Load(const char *fname)
+ConfigFile::ConfigFile(const std::string &fname) : filename(fname)
 {
 	ConfigSection *current_sect = nullptr;
 
-	this->Clear();
-
-	FILE *fp = fopen(fname, "rb");
-	if (fp == nullptr) return false;
+	FILE *fp = fopen(fname.c_str(), "rb");
+	if (fp == nullptr) return;
 
 	for (;;) {
 		char buffer[256];
 		char *line = fgets(buffer, lengthof(buffer), fp);
 		if (line == nullptr) break;
 
-		while (*line != '\0' && IsSpace(*line)) line++;
-		if (*line == '\0' || *line == ';') continue; // Silently skip empty lines or comment lines.
+		while (*line != '\0' && isspace(*line)) line++;
+		if (*line == '\0' || *line == ';' || *line == '#') continue; // Silently skip empty lines or comment lines.
 
 		if (*line == '[') {
 			/* New section. */
 			char *line2 = line;
 			while (*line2 != '\0' && *line2 != ']') line2++;
 			if (*line2 == ']') {
 				*line2 = '\0';
-				/* XXX Look for an existing config section? */
-				current_sect = new ConfigSection(StripWhitespace(line + 1, line2));
-				this->sections.push_front(current_sect);
+				const std::string sect_name(StripWhitespace(line + 1, line2));
+				if (this->sections.count(sect_name) == 0) this->sections.emplace(sect_name,
+						std::unique_ptr<ConfigSection>(new ConfigSection(*this, sect_name)));
```

Comment:
This looks very scary, let's keep single lines without braces a single line.

In other words stick a `{` and `}` in there when the line wraps.